test/loadtime/payload/payload.go

```diff
@@ -37,22 +38,28 @@ func NewBytes(p *Payload) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	h := []byte(fmt.Sprintf("%X", b))
```

Comment:
If we convert the _whole_ payload to hex, won't the reporting tool not be able to interpret the payload?

I'd recommend just converting the padding to hex, using a similar approach to what you've used here:

```golang
func makePadding(size uint64) ([]byte, error) {
    b := make([]byte, (size/2)+1))
    _, err := rand.Read(b)
    if err != nil {
        return nil, err
    }
    p := ([]byte(hex.EncodeToString(b)))[:size]
    return p, nil
}
```

I suspect the `hex.EncodeToString` call will also be more efficient than `fmt.Sprintf`.

Reply:
Thinking about this now though, would the rest of the payload data also potentially contain `=` symbols? I suspect it would, if it's binary-encoded.

This means the payload would need to be encoded in a non-binary format, but how do we then reliably decode it using the reporting tool? (Both approaches we have now would result in us sometimes truncating the protobuf message)

Should we perhaps consider changing the payload format to be a plain text-based format? Perhaps just JSON with whitespace at the end for padding?

Reply:
Thanks for the `hex.EncodeToString` suggestion. I'll adapt the PR.

As for encoding only the padding, I thought about it, but that would still be problematic if the protobuf marshalling of our data (timestamp, uuid, size, etc) happens to contain a byte "=". That's why I ended up encoding everything, even if that means the min payload is now doubled in size.

Reply:
> If we convert the whole payload to hex, won't the reporting tool not be able to interpret the payload?

The PR also adapts the unmarshalling (in a symmetric way)

Reply:
Ah, I see, that's great then!